runtime/memory.c

```diff
@@ -425,6 +425,26 @@ static struct pool_block* get_pool_block(caml_stat_block b)
   }
 }
 
+/* Linking a pool block into the ring */
+static void link_pool_block(struct pool_block *pb)
+{
+  caml_plat_lock(&pool_mutex);
+  pb->next = pool->next;
+  pb->prev = pool;
+  pool->next->prev = pb;
+  pool->next = pb;
+  caml_plat_unlock(&pool_mutex);
+}
+
+/* Unlinking a pool block from the ring */
+static void unlink_pool_block(struct pool_block *pb)
+{
+    caml_plat_lock(&pool_mutex);
+    pb->prev->next = pb->next;
+    pb->next->prev = pb->prev;
+    caml_plat_unlock(&pool_mutex);
+}
```

Comment:
Note: after this function runs, `pb->next` and `pb->prev` still points to the old neighbor blocks, but `pb` would is not a valid in-pool block at least until it is relinked. Unless performance is critical here, I would feel nicer if we wrote `NULL` in `pb->next` and `pb->prev` to easily distinguish unlinked blocks.